lib/managers/files.js

```diff
@@ -93,6 +94,36 @@ function createFileContentFormData(content) {
 	};
 }
 
+/**
+ * Poll the representation info URL until representation is generated,
+ * then return content URL template.
+ * @param {BoxClient} client The client to use for making API calls
+ * @param {string} infoURL The URL to use for getting representation info
+ * @returns {Promise<string>} A promise resolving to the content URL template
+ */
+function pollRepresentationInfo(client, infoURL) {
+
+	return client.get(infoURL)
+		.then(response => {
+
+			if (response.statusCode !== 200) {
+				throw errors.buildUnexpectedResponseError(response);
+			}
+
+			var info = response.body;
+
+			if (info.status.state === 'error') {
+				throw errors.buildResponseError(response, 'Representation had error status');
+			}
+
+			if (info.status.state !== 'success') {
```

Comment:
This function should do the same thing as the switch statement below. Seems like a good place to extract a function and all it in both places. (I.e. this function doesn't check for `viewable`.